pkg/storage/gcs/gcp_labels_tags.go

```diff
@@ -170,120 +183,190 @@ func filterTagList(ctx context.Context, client *rscmgr.TagBindingsClient, parent
-	// here to 8 requests per second.
-	limiter := newLimiter(gcpTagsRequestRateLimit, gcpTagsRequestTokenBucketSize, true)
+func (c *tagBindingsClient) listEffectiveTags(ctx context.Context, name string) *rscmgr.EffectiveTagIterator {
+	return c.TagBindingsClient.ListEffectiveTags(ctx, &rscmgrpb.ListEffectiveTagsRequest{
+		Parent: name,
+	})
+}
 
-	tagBindingReq := &rscmgrpb.CreateTagBindingRequest{
+func (c *tagBindingsClient) createTagBinding(ctx context.Context, parent, tag string) (*rscmgr.CreateTagBindingOperation, error) {
+	op, err := c.TagBindingsClient.CreateTagBinding(ctx, &rscmgrpb.CreateTagBindingRequest{
 		TagBinding: &rscmgrpb.TagBinding{
-			Parent: parent,
+			Parent:                 parent,
+			TagValueNamespacedName: tag,
 		},
-	}
+	}, getTagCreateCallOptions()...)
+	return op, err
+}
+
+func (c *tagBindingsClient) wait(ctx context.Context, op *rscmgr.CreateTagBindingOperation) error {
+	_, err := op.Wait(ctx)
+	return err
+}
+
+// CreateTagBindings creates the tag bindings for the resource.
+func (c *tagBindingsClient) CreateTagBindings(ctx context.Context, parent string, tags []string) error {
+	// GCP has a rate limit of 600 requests per minute, restricting
+	// here to 8 requests per second.
+	limiter := newRequestLimiter(gcpTagsRequestRateLimit, gcpTagsRequestTokenBucketSize, true)
+
 	errFlag := false
-	for _, value := range tagValues {
+	for _, tag := range tags {
 		if err := limiter.Wait(ctx); err != nil {
 			errFlag = true
 			klog.Errorf("rate limiting request to add %s tag to %s bucket failed: %v",
-				value, bucketName, err)
+				tag, parent, err)
 			continue
 		}
 
-		tagBindingReq.TagBinding.TagValueNamespacedName = value
-		result, err := client.CreateTagBinding(ctx, tagBindingReq, getCreateCallOptions()...)
+		result, err := c.createTagBinding(ctx, parent, tag)
 		if err != nil {
-			e, ok := err.(*apierror.APIError)
-			if ok && e.HTTPCode() == http.StatusConflict {
-				klog.Infof("tag binding %s/%s already exists", bucketName, value)
+			var gErr *apierror.APIError
+			if errors.As(err, &gErr) && gErr.HTTPCode() == http.StatusConflict {
+				klog.V(5).Infof("tag %s already exist on %s", tag, parent)
 				continue
 			}
 			errFlag = true
-			klog.Errorf("request to add %s tag to %s bucket failed: %v", value, bucketName, err)
+			klog.Errorf("request to add %s tag to %s bucket failed: %v", tag, parent, err)
 			continue
 		}
 
-		if _, err = result.Wait(ctx); err != nil {
+		if err = c.wait(ctx, result); err != nil {
 			errFlag = true
-			klog.Errorf("failed to add %s tag to %s bucket: %v", value, bucketName, err)
+			klog.Errorf("failed to add %s tag to %s bucket: %v", tag, parent, err)
 		}
-		klog.V(1).Infof("binding tag %s to %s bucket successful", value, bucketName)
+		klog.Infof("successfully added %s tag to %s bucket", tag, parent)
 	}
 	if errFlag {
-		return fmt.Errorf("failed to add tag(s) to %s bucket", bucketName)
+		return fmt.Errorf("failed to add tag(s) to %s bucket", parent)
```

Comment:
what's the motivation to rename `bucketName` to `parent`? I thought `bucketName` was more clear 🤔

Reply:
gapi's refer the resource to which tag will be attached as parent, so thought to use the same. But yeah since currently it's just the bucket which is tagged and logs also mention has bucket, bucketName seems better. I will change it to bucketName and use parent in the wrapper method for gapi.

Reply:
I updated it to resourceName to keep it generic. Please let me know WDYT